depmapomics/upload.py

```diff
@@ -0,0 +1,613 @@
+        # pick the PR from the most prioritized source according to the
+        # ranking in source_priority
+        elif len(prs_in_model[prs_in_model.Datatype == "rna"]) > 1:
+            cds_ids = prs_in_model[prs_in_model.Datatype == "rna"].CDSID.tolist()
+            subset_seq_table = seq_table[seq_table.index.isin(cds_ids)]
+            subset_seq_table.source = subset_seq_table.source.replace(source_priority)
+            latest_cds_id = subset_seq_table.loc[cds_ids, "source"].idxmin()
+            pr = subset_pr_table[subset_pr_table.CDSID == latest_cds_id].index[0]
+            rows.append((m, pr, "rna"))
+        # dna
+        # if there is only one dna (wes + wgs) PR associated with this Model, pick that PR
+        if (
+            len(
+                prs_in_model[
+                    (prs_in_model.Datatype == "wgs") | (prs_in_model.Datatype == "wes")
+                ]
+            )
+            == 1
+        ):
+            pr = prs_in_model[
+                (prs_in_model.Datatype == "wgs") | (prs_in_model.Datatype == "wes")
+            ].index[0]
+            rows.append((m, pr, "dna"))
+        # if there are more than one PRs associated with this Model
+        elif (
+            len(
+                prs_in_model[
+                    (prs_in_model.Datatype == "wgs") | (prs_in_model.Datatype == "wes")
+                ]
+            )
+            > 1
+        ):
+            cds_ids_wgs = prs_in_model[prs_in_model.Datatype == "wgs"].CDSID.tolist()
+            cds_ids_wes = prs_in_model[
+                (prs_in_model.Datatype == "wes") & (prs_in_model.CDSID != "")
+            ].CDSID.tolist()  # CDSID is '' when the profile is in legacy
+            pr = ""
+            # if no wgs, look at MC table and select the most prioritized source
+            # according to the ranking in source_priority
+            if len(cds_ids_wgs) == 0:
+                subset_seq_table = seq_table[seq_table.index.isin(cds_ids_wes)]
+                subset_seq_table.source = subset_seq_table.source.replace(
+                    source_priority
+                )
+                latest_cds_id_wes = subset_seq_table.loc[cds_ids_wes, "source"].idxmin()
+                pr = subset_pr_table[subset_pr_table.CDSID == latest_cds_id_wes].index[
+                    0
+                ]
+            # if there is wgs, always select wgs
+            else:
+                subset_seq_table = seq_table[seq_table.index.isin(cds_ids_wgs)]
+                subset_seq_table.source = subset_seq_table.source.replace(
+                    source_priority
+                )
+                latest_cds_id_wgs = subset_seq_table.loc[cds_ids_wgs, "source"].idxmin()
+                pr = subset_pr_table[subset_pr_table.CDSID == latest_cds_id_wgs].index[
+                    0
+                ]
+            rows.append((m, pr, "dna"))
+    default_table = pd.DataFrame(rows, columns=colnames)
+    return default_table
+
+
+def initVirtualDatasets(
```

Comment:
shouldn't they be all in the same dataset?

Reply:
we have different datasets for different portals. also this function only initializes the dataset by uploading a dummy file (taiga doesn't allow the creation of "empty" datasets)